syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.service";
option java_outer_classname = "ServiceProto";
option objc_class_prefix = "HLW";

package service;

service Service {
  rpc GreetingsUnaryUnary (GreetingsRequest) returns (GreetingsResponse) {}
  rpc GreetingsStreamUnary (stream GreetingsRequest) returns (GreetingsResponse) {}
  rpc GreetingsUnaryStream (GreetingsRequest) returns (stream GreetingsResponse) {}
  rpc GreetingsStreamStream (stream GreetingsRequest) returns (stream GreetingsResponse) {}

  rpc MultiplyUnaryUnary (MultiplyRequest) returns (MultiplyResponse) {}
  rpc MultiplyStreamUnary (stream MultiplyRequest) returns (MultiplyResponse) {}
  rpc MultiplyUnaryStream (MultiplyRequest) returns (stream MultiplyResponse) {}
  rpc MultiplyStreamStream (stream MultiplyRequest) returns (stream MultiplyResponse) {}

  rpc FibonacciUnaryUnary (FibonacciRequest) returns (FibonacciResponse) {}
  rpc FibonacciStreamUnary (stream FibonacciRequest) returns (FibonacciResponse) {}
  rpc FibonacciUnaryStream (FibonacciRequest) returns (stream FibonacciResponse) {}
  rpc FibonacciStreamStream (stream FibonacciRequest) returns (stream FibonacciResponse) {}
}

message GreetingsRequest {
  string name = 1;
}

message GreetingsResponse {
  string message = 1;
}

message MultiplyRequest {
  int32 number = 1;
  int32 multiplier = 2;
}

message MultiplyResponse {
  int64 result = 1;
}

message FibonacciRequest {
  int32 number = 1;
}

message FibonacciResponse {
  int64 result = 1;
}